src/nimblepkg/cli.nim

```diff
@@ -169,20 +175,124 @@ proc promptCustom*(forcePrompts: ForcePrompt, question, default: string): string
 proc promptCustom*(question, default: string): string =
   return promptCustom(dontForcePrompt, question, default)
 
+proc getKey(): Key =
+  ## Needed in order to allow arrow key control for Windows. It also makes it
+  ## easier to determine which keys are pressed on POSIX systems. This is based
+  ## of terminal.getch for the standard library.
+  when defined(windows):
+    let fd = getStdHandle(STD_INPUT_HANDLE)
+    var keyEvent = KEY_EVENT_RECORD()
+    var numRead: cint
+    while true:
+      # Block until character is entered
+      doAssert(waitForSingleObject(fd, INFINITE) == WAIT_OBJECT_0)
+      doAssert(readConsoleInput(fd, addr(keyEvent), 1, addr(numRead)) != 0)
+      if numRead == 0 or keyEvent.eventType != 1 or keyEvent.bKeyDown == 0:
+        continue
+      case keyEvent.wVirtualKeyCode:
+      of 0x26:
+        return kArrowUp
+      of 0x28:
+        return kArrowDown
+      of 0x0D:
+        return kEnter
+      else:
+        continue
+  else:
+    case getch()
+    of '\r':
+      return kEnter
+    of '\27':
+      if getch() == '[':
+        case getch():
+        of 'A':
+          return kArrowUp
+        of 'B':
+          return kArrowDown
+        else: discard
+    else: discard
+    result = getKey()
+
+proc promptListInteractive(question: string, args: openarray[string]) : string =
```

Comment:
Style nitpick: no space before ``:``.

Reply:
My preference as well. I guess I just hit the space bar an extra time